src/index.js

```diff
@@ -1,390 +1,583 @@
+    element,
+    lastNode,
+    nextNode,
+    lifecycle,
+    eventProxy
+  )
 
-      if (attributes.ondestroy) {
-        attributes.ondestroy(element)
-      }
-    }
-    return element
-  }
+  while (lifecycle.length > 0) lifecycle.pop()()
+
+  return element
+}
+
+export var h = function(name, props) {
+  var node
+  var rest = []
+  var children = []
+  var length = arguments.length
 
-  function removeElement(parent, element, node) {
-    function done() {
-      parent.removeChild(removeChildren(element, node))
+  while (length-- > 2) rest.push(arguments[length])
+
+  if ((props = props == null ? {} : props).children != null) {
+    if (rest.length <= 0) {
+      rest.push(props.children)
     }
+    delete props.children
+  }
 
-    var cb = node.attributes && node.attributes.onremove
-    if (cb) {
-      cb(element, done)
+  while (rest.length > 0) {
+    if (isArray((node = rest.pop()))) {
+      for (length = node.length; length-- > 0; ) {
+        rest.push(node[length])
+      }
+    } else if (node === false || node === true || node == null) {
     } else {
-      done()
+      children.push(typeof node === "object" ? node : createTextVNode(node))
     }
   }
 
-  function patch(parent, element, oldNode, node, isSvg) {
-    if (node === oldNode) {
-    } else if (oldNode == null || oldNode.nodeName !== node.nodeName) {
-      var newElement = createElement(node, isSvg)
-      parent.insertBefore(newElement, element)
-
-      if (oldNode != null) {
-        removeElement(parent, element, oldNode)
-      }
+  return typeof name === "function"
+    ? name(props, (props.children = children))
+    : createVNode(name, props, children, null, props.key, DEFAULT)
+}
 
-      element = newElement
-    } else if (oldNode.nodeName == null) {
-      element.nodeValue = node
-    } else {
-      updateElement(
-        element,
-        oldNode.attributes,
-        node.attributes,
-        (isSvg = isSvg || node.nodeName === "svg")
-      )
+var cancel = function(sub) {
+  sub.cancel()
+}
 
-      var oldKeyed = {}
-      var newKeyed = {}
-      var oldElements = []
-      var oldChildren = oldNode.children
-      var children = node.children
+var update = function(sub, oldSub, dispatch) {
+  for (var k in merge(sub, oldSub)) {
+    if (k !== "cancel" && sub[k] !== oldSub[k]) {
+      cancel(oldSub)
+      return start(sub, dispatch)
+    }
+  }
+  return oldSub
+}
 
-      for (var i = 0; i < oldChildren.length; i++) {
-        oldElements[i] = element.childNodes[i]
+var start = function(sub, dispatch) {
+  return merge(sub, {
+    cancel: sub.effect(sub, dispatch)
+  })
+}
 
-        var oldKey = getKey(oldChildren[i])
-        if (oldKey != null) {
-          oldKeyed[oldKey] = [oldElements[i], oldChildren[i]]
-        }
-      }
+var refresh = function(sub, oldSub, dispatch) {
+  if (isArray(sub) || isArray(oldSub)) {
+    var out = []
+    var subs = isArray(sub) ? sub : [sub]
+    var oldSubs = isArray(oldSub) ? oldSub : [oldSubs]
```

Comment:
I think there's an error with this line. In fact, the line should be `var oldSubs = isArray(oldSub) ? oldSub : [oldSub]` instead, no?

Reply:
@Seldszar Yes! Thanks for catching this. 🎉🎉